plugins/optimization-detective/optimization.php

```diff
@@ -80,12 +80,43 @@ static function ( string $output, ?int $phase ): string {
  * @access private
  */
 function od_maybe_add_template_output_buffer_filter(): void {
-	if (
-		! od_can_optimize_response() ||
-		od_is_rest_api_unavailable() ||
-		isset( $_GET['optimization_detective_disabled'] ) // phpcs:ignore WordPress.Security.NonceVerification.Recommended
-
-	) {
+	$conditions = array(
+		array(
+			'test'   => od_can_optimize_response(),
+			'reason' => __( 'Page is not optimized because od_can_optimize_response() returned false. This can be overridden with the od_can_optimize_response filter.', 'optimization-detective' ),
+		),
+		array(
+			'test'   => ! ( od_is_rest_api_unavailable() && ( wp_get_environment_type() !== 'local' || class_exists( 'Test_OD_Optimization' ) ) ),
```

Comment:
I wish the `class_exists( 'Test_OD_Optimization' )` check weren't needed, but there's apparently no way to override the return value of `wp_get_environment_type()` in tests.

Reply:
Not a blocker, but is there a more generic way to check if tests are being run? Maybe a constant that the WP core unit test suite always sets?